lib/native/mqtt.ts

```diff
@@ -419,7 +419,17 @@ export class MqttClientConnection extends NativeResourceMixin(BufferedEventEmitt
     private _reject(reject: (reason: any) => void) {
         return (reason: any) => {
             reject(reason);
-            this.emit('error', new CrtError(reason));
+
+            /*
+             * We are almost always rejecting promises via NAPI calls from C.  If we emit to an event
+             * with no handler, that leads to an exception that will cause native code to crash if invoked
+             * with AWS_NAPI_ENSURE.  Since error does not have a default handler attached during connection
+             * creation, check it first in order to avoid that behavior.
+             */
+            let listenerArray = this.rawListeners('error');
+            if (Array.isArray(listenerArray) && listenerArray.length > 0) {
+                this.emit('error', new CrtError(reason));
+            }
```

Comment:
Actually, from what I can find in other libraries, the way JS people handle this problem is:
```
process.nextTick(() => {
    this.emit('error, new CrtError(reason));
}
```

This will allow the current closure to return to native without exception, and then throw in node's JS context and find its way to the user's nearest exception handler.